File: sieve_of_eratosthenes/javascript/sieve.js
let sieveOfEratosthenes = function(n) {
  let array = [];
  let upperLimit = Math.sqrt(n);
  let output = [];

  for (let i = 0; i < n; i++) {
    array.push(true);
  }

  for (let i = 2; i <= upperLimit; i++) {
    if (array[i]) {
      for (let j = i * i; j < n; j += i) {
        array[j] = false;
      }
    }
  }

  for (let i = 2; i < n; i++) {
    if (array[i]) {
      output.push(i);
    }
  }

  return output;
};

File: algorithms/SieveEratosthenes/Sieve.kt
import kotlin.math.sqrt

/**
 * @created October 2, 2019
 * Sieve of Erastothenes is an algorithm to output all the prime numbers less than 10^6
 *
 * @reference https://en.wikipedia.org/wiki/Sieve_of_Eratosthenes
 */

private const val MAX = 1000

private fun sieve() {
    val prime = BooleanArray(MAX + 1)
    (2..MAX).forEach {
        prime[it] = true
    }
    (2..sqrt(MAX.toDouble()).toInt()).forEach { i ->
        if (prime[i]) {
            for (j in (i * i)..MAX step i) {
                prime[j] = false
            }
        }
    }
    println("All the prime numbers between 1 and $MAX:")
    (2..MAX).forEach {
        if (prime[it]) {
            print("$it ")
        }
    }
    println()
}

fun main() {
    sieve()
}